### PR TITLE
run osq runtime tests on windows

### PR DESCRIPTION
**Pull request by James-Pickett**
This PR enables osq runtime tests on windows. When running the tests locally on windows, I've only been able to get them to work when running as administrator. I think that's okay?

Thera are a few tests added to `runtime_posix_test.go` that would not run or were flakey on windows.

I spent a few hours trying to figure out why the `TestRestart` test would not work consistently on windows, but I was stumped. If anybody has an ideas there, it would be great.

### CODE REVIEW

File: pkg/osquery/runtime/runtime_all_test.go [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+func waitHealthy(t *testing.T, runner *Runner) {
+	require.NoError(t, backoff.WaitFor(func() error {
+		if runner.Healthy() == nil {
+			return nil
+		}
+		return fmt.Errorf("instance not healthy")
+	}, 120*time.Second, 5*time.Second))
+}
+
+func TestSimplePath(t *testing.T) {
```
Comment (James-Pickett) [inline pkg/osquery/runtime/runtime_all_test.go:102]: First goal is to get this test working on windows

File: pkg/osquery/runtime/runtime_all_test.go [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+		k,
+		WithKnapsack(k),
+		WithRootDirectory(rootDirectory),
+		WithOsquerydBinary(testOsqueryBinaryDirectory),
+	)
+
+	go runner.Run()
+
+	// never gets health on windows
+	waitHealthy(t, runner)
```
Comment (James-Pickett) [inline pkg/osquery/runtime/runtime_all_test.go:125]: never gets healthy

File: pkg/osquery/runtime/osqueryinstance.go [hunk 3]
```diff
@@ -592,6 +592,13 @@ func (o *OsqueryInstance) StartOsqueryExtensionManagerServer(name string, socket
...
 				"extension_name", name,
 			)
+			// PR Note, remove after test working
+			// On windows, during test this always returns error
+			// "status 1 registering extension: Failed adding registry: SQLITE_CANTOPEN"
+			// the source of the error is in osquery-go code
+			// https://github.com/osquery/osquery-go/blob/master/server.go#L217
+			// coming out of
+			// https://github.com/osquery/osquery-go/blob/master/gen/osquery/osquery.go#L2225
 			return fmt.Errorf("running extension server: %w", err)
```
Comment (James-Pickett) [inline pkg/osquery/runtime/osqueryinstance.go:602]: always hits error

```
"status 1 registering extension: Failed adding registry: SQLITE_CANTOPEN"
```

Comment (directionless) [reply pkg/osquery/runtime/osqueryinstance.go:602]: This is coming out of osquery. https://github.com/osquery/osquery/blob/dcd8594f8d8687cb13bc9540b5aad2c26c5f8d9e/osquery/extensions/interface.cpp#L147

Extensions register themselves to osquery, which sticks them into the registry. `SQLITE_CANTOPEN` smells a lot like some systemy permissiony issue. Do these tests work on a windows machine? (eg: is it a function of CI)

Comment (James-Pickett) [reply pkg/osquery/runtime/osqueryinstance.go:602]: it does not work on my windows machine, but it does work on CI .... I will investigate the permissions on my windows machine

Comment (James-Pickett) [reply pkg/osquery/runtime/osqueryinstance.go]: it works when you run as admin

File: pkg/osquery/runtime/runtime_posix_test.go [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+
+func requirePgidMatch(t *testing.T, pid int) {
+	pgid, err := syscall.Getpgid(pid)
+	require.NoError(t, err)
+	require.Equal(t, pgid, pid)
+}
+
+// TestOsquerySlowStart tests that the launcher can handle a slow-starting osqueryd process.
+// This this is only enabled on non-Windows platforms because suspending we have not yet
+// figured out how to suspend a process on windows via golang.
```
Comment (directionless) [inline pkg/osquery/runtime/runtime_posix_test.go:32]: (nit) grammar seems wrong. I think I know what you mean, but I'm not totally sure

File: pkg/osquery/runtime/runtime_test.go [hunk 5]
```diff
@@ -1,34 +1,29 @@
-//go:build !windows
-// +build !windows
-
 package runtime
 
+// these tests have to be run as admin on windows
+
```
Comment (directionless) [inline pkg/osquery/runtime/runtime_test.go:4]: Is it worth baking this into an `if` with a either an error or skip?

File: pkg/osquery/runtime/runtime_posix_test.go [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+	require.NoError(t, err)
+	assert.Equal(t, int32(0), resp.Status.Code)
+	assert.Equal(t, "OK", resp.Status.Message)
+
+	require.NoError(t, runner.Shutdown())
+}
+
+// TestRestart tests that the launcher can restart the osqueryd process.
+// This test causes time outs on windows, so it is only run on non-windows platforms.
+// Should investigate why this is the case.
```
Comment (directionless) [inline pkg/osquery/runtime/runtime_posix_test.go:123]: I wonder if this timeout is related to our prod issue, or unrelated. I think it's reasonable to refactor (as this PR does) and pick that up next PR

**Review by directionless — APPROVED** [review]
Nice refactor. I love bringing some of the tests into windows.

File: pkg/osquery/runtime/runtime_posix_test.go (hunk 6 above)
Comment (James-Pickett) [reply pkg/osquery/runtime/runtime_posix_test.go:123]: was thinking the same, merge this then try to dig into this test

File: pkg/osquery/runtime/runtime_test.go (hunk 5 above)
Comment (James-Pickett) [reply pkg/osquery/runtime/runtime_test.go:4]: I added it to the test main func, it just logs to stdout, the alternative was to added it to ever single test, which seemed tedious

File: pkg/osquery/runtime/runtime_posix_test.go (hunk 4 above)
Comment (James-Pickett) [reply pkg/osquery/runtime/runtime_posix_test.go]: updated

**Review by directionless — APPROVED** [review]

**Pull request merged**